homeassistant/components/stream/const.py

```diff
@@ -19,7 +19,11 @@
 
 NUM_PLAYLIST_SEGMENTS = 3  # Number of segments to use in HLS playlist
 MAX_SEGMENTS = 4  # Max number of segments to keep around
-MIN_SEGMENT_DURATION = 1.5  # Each segment is at least this many seconds
+TARGET_SEGMENT_DURATION = 2.0  # Each segment is about this many seconds
```

Comment:
Alternatively we could leave `TARGET_SEGMENT_DURATION` set at 1.5. This would keep this change from affecting behavior for those with frame intervals of 0.75 or 0.5 seconds.

Reply:
I think i have to admit I don't totally get the implications of doing this.  Performance vs latency trade off?

Reply:
Right now the worker finds the end of the segment by waiting until the `MIN_SEGMENT_DURATION` is reached (by looking at the pts) and then looking for the next video keyframe. If the keyframe exactly lines up with the `MIN_SEGMENT_DURATION`, the condition is satisfied and the segment ends, but if there is any slight adjustment to the pts that reduces the segment duration slightly, this condition will not be satisfied and we will have to wait until the next keyframe to end the segment. It's not catastrophic, but it makes more sense to just adjust the `MIN_SEGMENT_DURATION` slightly below what we are targeting so we don't risk missing the keyframe and we get segment durations which are closer to our target.
I had previously set this value to 1.5 as most cameras use a default keyframe interval of 1 second, so most people would end up seeing 2 second segment durations with this setting. However, @hunterjm noted that his segment durations end up being smaller, which means his keyframe interval is lower. (For example, if your keyframe interval is set to 0.8, the current behavior is to wait until 1.5 seconds is reached, then look for the next keyframe which comes at 1.6, so your segment durations would be 1.6).
If we make `TARGET_SEGMENT_DURATION` 2.0, the `MIN_SEGMENT_DURATION` would become 1.9 and people with keyframe intervals in the ranges [0.75,0.95) or [1.5,1.9) would experience slightly longer segments than before the change. If we make `TARGET_SEGMENT_DURATION` 1.5, the `MIN_SEGMENT_DURATION` would become 1.4, and this would only affect people with keyframe settings in the ranges [0.7,0.75) or [1.4,1.5) would experience slightly _shorter_ segments than before the change.
Basically, if we make it 2.0, a few people might notice a small difference, and if we make it 1.5, almost no people will notice a difference. I still prefer to make it 2.0, because it's a rounder number. The increase in latency can be mitigated by a small change to the manifest which I will introduce in the next PR or two.

Reply:
:thumbsup: